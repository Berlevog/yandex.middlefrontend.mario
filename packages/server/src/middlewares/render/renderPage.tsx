export const escape = (str: string) => {
	return str
		.replace(/&/g, "&amp;")
		.replace(/"/g, "&quot;")
		.replace(/'/g, "&#39;")
		.replace(/</g, "&lt;")
		.replace(/>/g, "&gt;");
};

type RenderPageParms = {
	reactHtml: string, reduxState: string, css: string, nonce: string, meta: any
}

export function renderPage({ meta, css, reactHtml, reduxState, nonce }:RenderPageParms) {
	return `
        <!DOCTYPE html>
        <html lang="en">
          <head>
              <meta charset="utf-8" />
              <meta content="width=device-width, initial-scale=1" name="viewport" />
              <meta content="#000000" name="theme-color" />
              <meta content="Mario game" name="description" />
              <link href="manifest.json" rel="manifest" />
              <link href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&family=VT323&display=swap" rel="stylesheet" />
              <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet" />
              <title>Mario Pro Max | SSR</title>
              <meta property="csp-nonce" content="${nonce}">
              ${meta}
              <style id="jss-server-side" nonce="${nonce}">${css}</style>
          </head>
          <body>
              <div id="root">${reactHtml}</div>
              <script nonce="${nonce}" id="initial-data" type="text/plain" data-json="${escape(reduxState)}"></script>
              <script defer src="/runtime~main.js" nonce="${nonce}"></script>
              <script defer src="/main.js" nonce="${nonce}"></script>
          </body>
        </html>
    `;
}
