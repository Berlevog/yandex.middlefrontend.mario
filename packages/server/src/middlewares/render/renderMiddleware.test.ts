import { renderPage } from './renderPage';

test('should create safe html', () => {
  const meta = '<meta>meta</meta>';
  const css = '<style>style</style>';
  const reactHtml = '<body>body</body>';
  const reduxState = JSON.stringify({
    arr: ['foo', 'bar'],
    obj: {
      foo: 1,
      bar: 1,
    },
    str: 'foo bar',
  });
  // const scripts = '<script>window</script>';
  const nonce = 'nonce';

  expect(
    renderPage({ meta, css, reactHtml, reduxState, nonce })
  ).toMatchSnapshot();
});
