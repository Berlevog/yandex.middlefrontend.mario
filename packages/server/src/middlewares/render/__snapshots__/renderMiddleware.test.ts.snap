// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create safe html 1`] = `
"
        <!DOCTYPE html>
        <html lang=\\"en\\">
          <head>
              <meta charset=\\"utf-8\\" />
              <meta content=\\"width=device-width, initial-scale=1\\" name=\\"viewport\\" />
              <meta content=\\"#000000\\" name=\\"theme-color\\" />
              <meta content=\\"Mario game\\" name=\\"description\\" />
              <link href=\\"manifest.json\\" rel=\\"manifest\\" />
              <link href=\\"https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&family=VT323&display=swap\\" rel=\\"stylesheet\\" />
              <link href=\\"https://fonts.googleapis.com/icon?family=Material+Icons\\" rel=\\"stylesheet\\" />
              <title>Mario Pro Max | SSR</title>
              <meta property=\\"csp-nonce\\" content=\\"nonce\\">
              <meta>meta</meta>
              <style id=\\"jss-server-side\\" nonce=\\"nonce\\"><style>style</style></style>
          </head>
          <body>
              <div id=\\"root\\"><body>body</body></div>
              <script nonce=\\"nonce\\" id=\\"initial-data\\" type=\\"text/plain\\" data-json=\\"{&quot;arr&quot;:[&quot;foo&quot;,&quot;bar&quot;],&quot;obj&quot;:{&quot;foo&quot;:1,&quot;bar&quot;:1},&quot;str&quot;:&quot;foo bar&quot;}\\"></script>
              <script defer src=\\"/runtime~main.js\\" nonce=\\"nonce\\"></script>
              <script defer src=\\"/main.js\\" nonce=\\"nonce\\"></script>
          </body>
        </html>
    "
`;
